target/linux/mpc85xx/files/arch/powerpc/boot/dts/t30-w.dts

```diff
@@ -0,0 +1,384 @@
+#include <dt-bindings/input/input.h>
+
+/include/ "fsl/p1020si-pre.dtsi"
+/ {
+	model = "watchguard,t30-w";
+	compatible = "watchguard,t30-w";
+	
+	aliases {
+		serial0 = &serial0;
+		serial1 = &serial1;
+		ethernet0 = &enet0;
+		pci0 = &pci0;
+		pci1 = &pci1;
+		led-boot = &attn_orange;
+		led-failsafe = &status_red;
+		led-running = &attn_orange;
+		led-upgrade = &status_red;
+	};
+		
+	memory {
+		device_type = "memory";
+	};
+
+	lbc: localbus@ffe05000 {
+		reg = <0x0 0xffe05000 0x0 0x1000>;
+		compatible = "fsl,p1020-elbc","fsl,elbc","simple-bus";
+		interrupts = <0x13 0x02>;
+		interrupt-parent = <&mpic>;
+
+		/* NOR Flash */
+		ranges = <0x00 0x00 0x00 0xefc00000 0x100000>;
+		
+		flash@0 {
+			#address-cells = <0x01>;
+			#size-cells = <0x01>;
+			compatible = "cfi-flash";
+			reg = <0x00 0x00 0x400000>;
+			bank-width = <0x02>;
+			device-width = <0x01>;
+
+			partition@0 {
+				reg = <0x00 0x20000>;
+				label = "nor-cfg0";
+			};
+
+			partition@20000 {
+				reg = <0x20000 0x10000>;
+				label = "nor-cfg1";
+			};
+
+			partition@30000 {
+				reg = <0x30000 0x20000>;
+				label = "nor-mfg";
+			};
+
+			partition@50000 {
+				reg = <0x50000 0xc0000>;
+				label = "nor-bootopt";
+			};
+
+			partition@110000 {
+				reg = <0x110000 0xc0000>;
+				label = "nor-extra1";
+			};
+
+			partition@1D0000 {
+				reg = <0x1d0000 0xc0000>;
+				label = "nor-extra2";
+			};
+
+			partition@290000 {
+				reg = <0x290000 0xe0000>;
+				label = "nor-backup";
+			};
+
+			partition@370000 {
+				reg = <0x370000 0x10000>;
+				label = "nor-ubootenv";
+			};
+
+			partition@380000 {
+				reg = <0x380000 0x80000>;
+				label = "nor-uboot";
+			};
+		};
+	};
+
+	soc: soc@ffe00000 {
+		compatible = "fsl,p1020-immr","simple-bus";
+		ranges = <0x0 0x0 0xffe00000 0x100000>;
+		
+		i2c@3000 {
+			rtc@30 {
+				compatible = "seiko,s35390a";
+				reg = <0x30>;
+			};
+		};
+
+		i2c@3100 {
+			tpm@29 {
+				compatible = "tpm,tpm_i2c_atmel";
+				reg = <0x29>;
+			};
+		};
+		
+		sdhc@2e000 {
+			compatible = "fsl,p1020-esdhc","fsl,esdhc";
+			sdhci,auto-cmd12;
+			reg = <0x2e000 0x1000>;
+			interrupts = <0x48 0x2 0x0 0x0>;
+			interrupt-parent = <&mpic>;
+			fsl,sdhci-dma-broken;
+			fsl,sdhci-ahb2mag-irq-bypass;
+			fsl,sdhci-adjust-timeout;
+			clock-frequency = <0>;
+		};
+
+		spi@7000 {
+			cell-index = <0x00>;
+			#address-cells = <0x01>;
+			#size-cells = <0x00>;
+			compatible = "fsl,mpc8536-espi";
+			reg = <0x7000 0x1000>;
+			interrupts = <0x3b 0x02>;
+			interrupt-parent = <&mpic>;
+			fsl,espi-num-chipselects = <0x04>;
+
+			flash@0 {
+				#address-cells = <0x01>;
+				#size-cells = <0x01>;
+				compatible = "everspin,mr25h256","jedec,spi-nor";
+				reg = <0>; /* Chip select 0 */
+				spi-max-frequency = <40000000>;
+				m25p,fast-read;
+
+				partition@spi_mtd_0 {
+					label = "spi_mtd_0";
+					reg = <0x00 0x8000>;
+				};
+			};
+
+			legerity@1 {
+				compatible = "zarlink,le88266";
+				reg = <0x01>;
+				spi-max-frequency = <0x7a120>;
```

Comment:
Ok, this tells me that you are probably using the "get a working fdt with all the uboot changes" and export it as the dts :) .  

This is because "500000" rolls much better off the keyboard (same as with the 40MHz from the node above).

Reply:
Good point, will change this